js/templates/post.mjs

```diff
@@ -0,0 +1,191 @@
+import * as post from '../api/posts/index.mjs';
+import { timeSince } from '../api/utilities/timeSince.mjs';
+import { removePost } from '../api/posts/delete.mjs';
+import { editPost } from '../api/posts/update.mjs';
+import { reactionToPost } from '../api/posts/react.mjs';
+
+function appendChildren(parent, children) {
+  children.forEach((child) => {
+    if (child) parent.appendChild(child);
+  });
+}
+
+export function postTemplate(postData) {
+  const post = document.createElement('div');
+  post.className = 'card border-dark mb-3 mx-auto position-relative';
+
+  // like icon //
+
+  const likeIcon = document.createElement('i');
+  likeIcon.className = 'bi bi-heart like-icon pointer-cursor';
+  likeIcon.textContent = postData.reactions.length;
+  likeIcon.addEventListener('click', async (event) => {
+    const response = await reactionToPost(postData.id);
+    if (response.ok) {
+      console.log('reaction saved');
+    } else {
+      console.error('reaction not saved');
+    }
+    setTimeout(() => {
+      location.reload();
+    }, 300);
+  });
+
+  // Thrashcan icon //
+
+  const trashIcon = document.createElement('i');
+
+  const authorName = postData.author.name;
+  const storageName = localStorage.getItem('profile');
+  const parsedName = JSON.parse(storageName);
+
+  const loggedInUserName = parsedName ? parsedName.name : null;
+
+  if (authorName !== loggedInUserName) {
+    trashIcon.style.display = 'none';
+  } else {
+    trashIcon.style.display = 'block';
+  }
+  trashIcon.className = 'bi bi-trash3 trash-icon pointer-cursor';
+  trashIcon.addEventListener('click', async (event) => {
+    try {
+      const response = await removePost(postData.id);
+      if (response.ok) {
+        console.log('post deleted');
+        post.remove();
+      } else {
+        console.log('post not deleted');
```

Comment:
`console.log()`'s should be removed from final code - using an `alert()` function is better in this instance.